src/main/java/com/teragrep/pth_06/ArchiveMicroStreamReader.java

```diff
@@ -187,24 +263,52 @@ public void stop() {
      */
     @Override
     public Offset latestOffset() {
+        // After rebase is complete: Refactor the DatasourceOffset and SerializedDatasourceOffset if the 8x else-if statements are too much.
```

Comment:
same here about the if-else

Reply:
Solved in commit 03552980d1289068474c778d099b2d0dc979c93f